File: FMAplication/Domain/DailyActivities/DailyAudit.cs
using FMAplication.Core;
using FMAplication.Domain.Reports;
using FMAplication.Enumerations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.DailyActivities
{

    public class DailyAudit : AuditableEntity<int>
    {
        public int DailyCMActivityId { get; set; }
        public bool IsDistributionCheck { get; set; }
        public bool IsFacingCount { get; set; }
        public bool IsPlanogramCheck { get; set; }
        public bool IsPriceAudit { get; set; }
        public int DistributionCheckStatus { get; set; }
        public int FacingCountStatus { get; set; }
        public int PlanogramCheckStatus { get; set; }
        public int PriceAuditCheckStatus { get; set; }
        [StringLength(512)]
        public string DistributionCheckIncompleteReason { get; set; }
        [StringLength(512)]
        public string FacingCountCheckIncompleteReason { get; set; }
        [StringLength(512)]
        public string PlanogramCheckIncompleteReason { get; set; }
        [StringLength(512)]
        public string PriceAuditCheckIncompleteReason { get; set; }
        public List<AuditReport> AllProducts { get; set; } = new List<AuditReport>();
    }
}

File: FMAplication/ClientApp/src/app/Shared/Entity/Daily-Activity/daily-audit.ts

  export class DailyAudit{
    public id:number;
    public dailyCMActivityId:number;


    public isDistributionCheck: boolean;
    public distributionCheckProducts:any[];
    public distributionCheckIncompleteReason:string;
    public distributionCheckStatus:number;

    public isFacingCount: boolean;
    public facingCountProducts:any[];
    public facingCountCheckIncompleteReason:string;
    public facingCountStatus:number;

    public isPlanogramCheck: boolean;
    public planogramCheckProducts: any[];
    public planogramCheckIncompleteReason: string;
    public planogramCheckStatus: number;

    public isPriceAudit: boolean;
    public priceAuditProducts: any[];
    public priceAuditCheckIncompleteReason: string;
    public priceAuditCheckStatus: string;

  }